### PR TITLE
win_nssm: strict fix + added dependencies,startmode,user/pw feature

### PR DESCRIPTION
**Pull request by h0nIg**
- added fix to fix strict variable problems
- added feature for minimum required parameters (dependencies, startmode, dedicated user for service startup)

### CODE REVIEW

**Comment by h0nIg** [comment]
could you please provide feedback? @schwartzmx @trondhindenes @petemounce @elventear @smadam813 @jhawkesworth @angstwad @sivel @chrishoffman @cchurch @georgefrank

**Comment by robynbergeron** [comment]
Hi @h0nIg -- thanks for this PR.

@smadam813 @georgefrank -- Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

Looks like the rest of the windows crew has already been hailed :)

File: windows/win_nssm.ps1 [hunk 1]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
 # POWERSHELL_COMMON
 
 $params = Parse-Args $args;
+
 $result = New-Object PSObject;
 Set-Attr $result "changed" $false;
 
+$name = Get-Attr $params "name" -failifempty $true
+$state = Get-Attr $params "state" "present" -validateSet "present", "absent", "started", "stopped", "restarted" -resultobj $result
```
Comment (smadam813) [inline windows/win_nssm.ps1]: I would change this to:
```
$state = Get-Attr $params "state" -validateSet "present", "absent", "started", "stopped", "restarted" -default "present" -resultobj $stateResult
```
That way you are explicitly stating the result object and the default state.

File: windows/win_nssm.ps1 [hunk 2]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
-
-If ($params.application)
-{
-    $application = $params.application
-}
-Else
-{
-    $application = $null
-}
+$application = Get-Attr $params "application" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$application = Get-Attr $params "application" -default $null
```

File: windows/win_nssm.ps1 [hunk 3]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
+$appParameters = Get-Attr $params "app_parameters" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$appParameters = Get-Attr $params "app_parameters" -default $null
```

File: windows/win_nssm.ps1 [hunk 4]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
+$startMode = Get-Attr $params "start_mode" "auto" -validateSet "auto", "manual", "disabled" -resultobj $result
```
Comment (smadam813) [inline windows/win_nssm.ps1]: I would change this to:
```
$startMode = Get-Attr $params "start_mode" -validateSet "auto", "manual", "disabled" -default "auto" -resultobj $startModeResult
```
That way you are explicitly stating the result object and the default state.

File: windows/win_nssm.ps1 [hunk 5]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
 
-If ($params.app_parameters)
-{
-    $appParameters = $params.app_parameters
-}
-Else
-{
-    $appParameters = $null
-}
+$stdoutFile = Get-Attr $params "stdout_file" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$stdoutFile = Get-Attr $params "stdout_file" -default $null
```

File: windows/win_nssm.ps1 [hunk 6]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
+$stderrFile = Get-Attr $params "stderr_file" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$stderrFile = Get-Attr $params "stderr_file" -default $null
```

File: windows/win_nssm.ps1 [hunk 7]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
+$dependencies = Get-Attr $params "dependencies" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$dependencies = Get-Attr $params "dependencies" -default $null
```

File: windows/win_nssm.ps1 [hunk 8]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
-
-If ($params.stderr_file)
-{
-    $stderrFile = $params.stderr_file
-}
-Else
-{
-    $stderrFile = $null
-}
+$user = Get-Attr $params "user" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$user = Get-Attr $params "user" -default $null
```

File: windows/win_nssm.ps1 [hunk 9]
```diff
@@ -22,69 +24,23 @@ $ErrorActionPreference = "Stop"
...
+$password = Get-Attr $params "password" $null
```
Comment (smadam813) [inline windows/win_nssm.ps1]: Explicit Default:
```
$password = Get-Attr $params "password" -default $null
```

**Comment by smadam813** [comment]
@h0nIg 
Long story short I will test in a bit and it looks good so far!  I would change some of the new functionality to use cmdlets vs positional parameters that way if they are ever changed, the script won't break.  I will give the module a test soon and have feedback on functionality.

**Comment by smadam813** [comment]
I created another PR (#1143) and merged your branch in.  Thank you very much for the PR @h0nIg !  I wanted to fix some issues I found and thought it might be tedious communicating them so I just took care of it.  @h0nIg all your commits will still be referenced and recognition will still be received :-)

You can see my changes here:
https://github.com/smadam813/ansible-modules-extras/commit/56e7d80479b9ac11eb75f6885b417f225bbb2caa

**Comment by gregdek** [comment]
@h0nIg thanks for the PR. Since @smadam813 is the owner of the module and brought it in through the other PR, we'll close this one and mark that one for merge.

**Pull request closed without merging**